diff --git a/bfd/coff-aarch64.c b/bfd/coff-aarch64.c
--- a/bfd/coff-aarch64.c
+++ b/bfd/coff-aarch64.c
@@ -531,7 +531,8 @@ coff_aarch64_relocate_section (bfd *output_bfd,
 
       h = obj_coff_sym_hashes (input_bfd)[symndx];
 
-      if (h && h->root.type == bfd_link_hash_defined)
+      if (h && (h->root.type == bfd_link_hash_defined
+		|| h->root.type == bfd_link_hash_defweak))
 	{
 	  sec = h->root.u.def.section;
 	  sym_value = h->root.u.def.value;
