diff --git a/gcc/config/mingw/winnt.cc b/gcc/config/mingw/winnt.cc
index a09f578..ceefce9 100644
--- a/gcc/config/mingw/winnt.cc
+++ b/gcc/config/mingw/winnt.cc
@@ -876,6 +876,12 @@ struct seh_frame_state
   bool after_prologue;
   bool in_epilogue;
 
+#if defined (TARGET_AARCH64_MS_ABI)
+  /* Track large stack allocations materialized in temporary registers.  */
+  bool aarch64_stackalloc_valid[FIRST_PSEUDO_REGISTER];
+  HOST_WIDE_INT aarch64_stackalloc_value[FIRST_PSEUDO_REGISTER];
+#endif
+
   /* True if we are in the cold section.  */
   bool in_cold_section;
 };
@@ -1499,7 +1505,6 @@ seh_pattern_emit (FILE *f, struct seh_frame_state *seh, rtx pat)
 	src = SET_SRC (pat);
 	HOST_WIDE_INT increment = 0;
 	dest = SET_DEST (pat);
-	static HOST_WIDE_INT stack_alloc = 0;
 
 	switch (GET_CODE (dest))
 	  {
@@ -1537,8 +1542,13 @@ seh_pattern_emit (FILE *f, struct seh_frame_state *seh, rtx pat)
 		else if (seh->in_epilogue && dest == stack_pointer_rtx
 			 && XEXP (src, 0) == stack_pointer_rtx
 			 && REG_P (XEXP (src, 1))
-			 && REGNO (XEXP (src, 1)) == R12_REGNUM)
-		  emitted = seh_emit_stackalloc (f, seh, stack_alloc);
+			 && seh->aarch64_stackalloc_valid[REGNO (XEXP (src, 1))])
+		  {
+		    unsigned int regno = REGNO (XEXP (src, 1));
+		    emitted = seh_emit_stackalloc (f, seh,
+						   seh->aarch64_stackalloc_value[regno]);
+		    seh->aarch64_stackalloc_valid[regno] = false;
+		  }
 		else if (dest == stack_pointer_rtx
 		    && CONST_INT_P (XEXP (src, 1)))
 		  emitted = seh_emit_stackalloc (f, seh,
@@ -1563,14 +1573,22 @@ seh_pattern_emit (FILE *f, struct seh_frame_state *seh, rtx pat)
 		extend = SET_SRC (pat);
 		extend = XEXP (extend, 1);
 		if (dest == stack_pointer_rtx && src == stack_pointer_rtx
-		    && REG_P (extend) && REGNO (extend) == R12_REGNUM)
-		  emitted = seh_emit_stackalloc (f, seh, stack_alloc);
+		    && REG_P (extend)
+		    && seh->aarch64_stackalloc_valid[REGNO (extend)])
+		  {
+		    unsigned int regno = REGNO (extend);
+		    emitted = seh_emit_stackalloc (f, seh,
+						   seh->aarch64_stackalloc_value[regno]);
+		    seh->aarch64_stackalloc_valid[regno] = false;
+		  }
    	      break;
 
 	      case CONST_INT:
-		increment = INTVAL (src);
-		if (REGNO(dest) == R12_REGNUM)
-		  stack_alloc = increment;
+		if (GP_REGNUM_P (REGNO (dest)))
+		  {
+		    seh->aarch64_stackalloc_valid[REGNO (dest)] = true;
+		    seh->aarch64_stackalloc_value[REGNO (dest)] = INTVAL (src);
+		  }
 		break;
 
 	      case MEM:
@@ -1586,6 +1604,8 @@ seh_pattern_emit (FILE *f, struct seh_frame_state *seh, rtx pat)
 		break;
 
 	      default:
+		if (GP_REGNUM_P (REGNO (dest)))
+		  seh->aarch64_stackalloc_valid[REGNO (dest)] = false;
 		break;
 	      }
 	    break;
