--- a/mingw-w64-headers/include/psdk_inc/intrin-impl.h
+++ b/mingw-w64-headers/include/psdk_inc/intrin-impl.h
@@ -1358,7 +1358,7 @@ __MINGW_EXTENSION __int64 _InterlockedExchange64(__int64 volatile *Target, __in
 #if !__has_builtin(_InterlockedExchange64)
 __MINGW_EXTENSION __INTRINSICS_USEINLINE 
 __int64 _InterlockedExchange64(__int64 volatile *Target, __int64 Value) {
-    return __sync_lock_test_and_set(Target, Value);
+    return __atomic_exchange_n(Target, Value, __ATOMIC_SEQ_CST);
 }
 #endif
 #define __INTRINSIC_DEFINED__InterlockedExchange64
@@ -1735,7 +1735,7 @@ __LONG32 _InterlockedExchange(__LONG32 volatile *Target, __LONG32 Value);
 #if !__has_builtin(_InterlockedExchange)
 __INTRINSICS_USEINLINE 
 __LONG32 _InterlockedExchange(__LONG32 volatile *Target, __LONG32 Value) {
-    return __sync_lock_test_and_set(Target, Value);
+    return __atomic_exchange_n(Target, Value, __ATOMIC_SEQ_CST);
 }
 #endif
 #define __INTRINSIC_DEFINED__InterlockedExchange
@@ -1768,7 +1768,7 @@ void *_InterlockedExchangePointer(void *volatile *Target,void *Value);
 #if !__has_builtin(_InterlockedExchangePointer)
 __INTRINSICS_USEINLINE 
 void *_InterlockedExchangePointer(void *volatile *Target,void *Value) {
-    return __sync_lock_test_and_set(Target, Value);
+    return __atomic_exchange_n(Target, Value, __ATOMIC_SEQ_CST);
 }
 #endif
 #define __INTRINSIC_DEFINED__InterlockedExchangePointer
