--- a/test/recipes/20-test_dhparam.t
+++ b/test/recipes/20-test_dhparam.t
@@ -221,4 +221,9 @@ my $inout = "inout.pem";
 copy($input, $inout);
 ok(run(app(['openssl', 'dhparam', '-in', $inout, '-out', $inout])),
     "identical infile and outfile");
-ok(!compare_text($input, $inout), "converted file $inout did not change");
+ok(!compare_text($input, $inout, sub {
+        my ($left, $right) = @_;
+        $left =~ s/\r\n/\n/g;
+        $right =~ s/\r\n/\n/g;
+        return $left ne $right;
+    }), "converted file $inout did not change");
