--- a/dist/aclocal/mutex.m4
+++ b/dist/aclocal/mutex.m4
@@ -496,7 +496,7 @@
 # ARM64/gcc: Linux
 if test "$db_cv_mutex" = no; then
 AC_TRY_COMPILE(,[
-#if defined(__arm64__) && defined(__GNUC__)
+#if (defined(__arm64__) || defined(__aarch64__)) && defined(__GNUC__)
 	return (0);
 #else
 	FAIL TO COMPILE/LINK
--- a/dist/configure
+++ b/dist/configure
@@ -25354,7 +25354,7 @@
 main (void)
 {
 
-#if defined(__arm64__) && defined(__GNUC__)
+#if (defined(__arm64__) || defined(__aarch64__)) && defined(__GNUC__)
 	return (0);
 #else
 	FAIL TO COMPILE/LINK
