--- a/configure.ac
+++ b/configure.ac
@@ -19,6 +19,7 @@
 AC_INIT([libiconv], [1.19])
 AC_CONFIG_SRCDIR([lib/iconv.c])
 AC_CONFIG_AUX_DIR([build-aux])
+AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE
 AC_CONFIG_HEADERS([config.h lib/config.h])
 AC_PROG_MAKE_SET
--- a/libcharset/configure.ac
+++ b/libcharset/configure.ac
@@ -19,6 +19,7 @@
 AC_INIT([libcharset], [1.5])
 AC_CONFIG_SRCDIR([lib/localcharset.c])
 AC_CONFIG_AUX_DIR([build-aux])
+AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS([config.h])
 AC_PROG_MAKE_SET
 
